core-services/ipi-deprovision/aws.sh

```diff
@@ -17,7 +17,11 @@ function queue() {
 
 function deprovision() {
   WORKDIR="${1}"
+  REGION="$(cat ${WORKDIR}/metadata.json|jq .aws.region -r)"
+  INFRA_ID="$(cat ${WORKDIR}/metadata.json|jq '.aws.identifier[0]|keys[0]' -r|cut -d '/' -f3|tr -d '\n')"
   timeout --signal=SIGQUIT 30m openshift-install --dir "${WORKDIR}" --log-level error destroy cluster && touch "${WORKDIR}/success" || touch "${WORKDIR}/failure"
+  timeout --signal=SIGQUIT 30m hypershift destroy infra aws --aws-creds "${AWS_SHARED_CREDENTIALS_FILE}" --infra-id "${INFRA_ID}" --region "${REGION}" || touch "${WORKDIR}/failure"
+  timeout --signal=SIGQUIT 30m hypershift destroy iam aws --aws-creds "${AWS_SHARED_CREDENTIALS_FILE}" --infra-id "${INFRA_ID}" --region "${REGION}" || touch "${WORKDIR}/failure"
```

Comment:
@wking is this ok? It will then also be used in the normal ipi-deprovision job but I don't think this will cause any issues...?

Reply:
Won't this lead to us exiting one with `Deprovision failed on the following clusters` down below for non-HyperShift clusters?  I guess I don't know which exit codes `hypershift destroy ...` uses for "hmm, the thing you asked me to delete isn't a cluster I know about, and I can't find any resources like that in AWS either".

Reply:
I just tested this manually and it does rc0 when given a non-existant infra id:
```
$ ./bin/hypershift destroy infra aws --aws-creds ~/.aws/credentials  --name=alvaro-test --base-domain ci.devcluster.openshift.com  --region=ca-central-1 --infra-id=nonsense; echo $?
INFO[0005] Successfully destroyed infrastructure        
0
```

```
$ ./bin/hypershift destroy iam aws --aws-creds ~/.aws/credentials  --region=ca-central-1 --infra-id=nonsense; echo $?
INFO[0000] Successfully destroyed IAM infra             
0
```

But I noticed it requires `--base-domain` or it will error, let me check if we can improve that.
/hold

Reply:
Okay, it doesn't error if there is no route 53 zone for the passed base domain, so I just made it default to what we use for hypershift with the option to override:

```
$ ./bin/hypershift destroy infra aws --aws-creds ~/.aws/credentials  --region=ca-central-1 --infra-id=nonsense --base-domain=definitely-not.there; echo $?
INFO[0006] Successfully destroyed infrastructure        
0
```

Reply:
@wking ptal if you can

Reply:
/lgtm